astropy/coordinates/sky_coordinate.py

```diff
@@ -782,6 +815,87 @@ def spherical_offsets_to(self, tocoord):
         dlat = acoord.spherical.lat.view(Angle)
         return dlon, dlat
 
+    def directional_offsets_to(self, tocoord):
+        r"""
+        Computes direction and distance to go *from* this coordinate *to* another.
+
+        Parameters
+        ----------
+        tocoord : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+            The coordinate to find the offset to.
+
+        Returns
+        -------
+        position_angle : `~astropy.coordinates.Angle`
+            The (positive) position angle of the vector pointing from ``self``
+            to ``tocoord``.
+        separation : `~astropy.coordinates.Angle`
+            The distance between ``self`` and ``tocoord``.
+
+        If either ``self`` or ``tocoord`` contain arrays, these will be a
+        arrays following the appropriate `numpy` broadcasting rules.
+
+        See Also
+        --------
+        position_angle : for the postion_angle component.
+        separation : for the angular offset component.
+        directional_offset_by : use offset to go from a coordinate to a new coordinate.
+
+        """
+
+        if self.is_equivalent_frame(tocoord):
+            other_in_self_frame = tocoord
+        else:
+            other_in_self_frame = tocoord.frame.transform_to(self.frame)
+
+        slat = self.represent_as(UnitSphericalRepresentation).lat
+        slon = self.represent_as(UnitSphericalRepresentation).lon
+        olat = other_in_self_frame.represent_as(UnitSphericalRepresentation).lat
+        olon = other_in_self_frame.represent_as(UnitSphericalRepresentation).lon
+
+        posang = angle_utilities.position_angle(slon, slat, olon, olat)
+        separation = angle_utilities.angular_separation(slon, slat, olon, olat)
+        return Angle(posang), Angle(separation)
+
+    def directional_offset_by(self, position_angle, separation):
+        r"""
+        Computes coordinates at the given offset from this coordinate.
+
+        Parameters
+        ----------
+        All parameters are `~astropy.coordinates.Angle` or float,
+        where float values are radians.
+
+        position_angle : position_angle of offset.
+        separation : offset distance.
+
+        Returns
+        -------
+        newpoints : SkyCoord offset by the given values.
+            `numpy` broadcasting rules apply if self or any arguments are arrays.
+
+        Notes
+        -----
+        Returned SkyCoord frame retains only those parameters specific
```

Comment:
The preferred behavior would be to have the returned SkyCoordFrame be the same as the input `self` frame, which is what I would naively expect from
`result = SkyCoord(newlon, newlat, frame=self.frame)`. 

For this specific PR I encapsulate the unintuitive behavior in a Note instead of trying to fix everything at once.  When the framework supports retaining frame information, we can fix it up and change the Note.  Since this PR won't be in the release until June, this behavior need not be petrified in.